src/extra/widgets/animbtn/lv_animbtn.c

```diff
@@ -0,0 +1,380 @@
+/**
+ * @file lv_animbtn.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_animbtn.h"
+
+#if LV_USE_ANIMBTN != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+#define MY_CLASS &lv_animbtn_class
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static void lv_animbtn_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj);
+static void lv_animbtn_destructor(const lv_obj_class_t * class_p, lv_obj_t * obj);
+static void lv_animbtn_event(const lv_obj_class_t * class_p, lv_event_t * e);
+static void apply_state(lv_obj_t * animbtn, bool skip_transition);
+static void loop_state(lv_obj_t * animbtn);
+static lv_animbtn_state_t suggest_state(lv_obj_t * animbtn, lv_animbtn_state_t state);
+static lv_animbtn_state_t get_state(const lv_obj_t * animbtn);
+static void setup_anim(lv_animbtn_t * animbtn, lv_animbtn_state_desc_t * desc);
+static int is_state_valid(const lv_animbtn_state_desc_t * state);
+static uint8_t find_trans(lv_animbtn_t * animbtn, lv_animbtn_state_t from, lv_animbtn_state_t to);
+static bool is_transiting(lv_animbtn_t * animbtn, lv_animbtn_state_t current_state);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+const lv_obj_class_t lv_animbtn_class = {
+    .base_class = &lv_obj_class,
```

Comment:
Can is be based on `lv_img` and have functions like  `lv_animbtn_create(parent)` and `lv_animbtn_set_src()` (wrapper to `lv_ing_set_src`)?

Reply:
Yes that's possible. `lv_animbtn_set_src` will be useless if the first element of the struct is a `lv_img_t`, one can use `lv_img_set_src` directly then.

In that case, I think it's better to use lv_img's API directly, since one my want to change the size, etc... and I don't want to replicate the complete API with `lv_anim_` prefix.